src/plot_api/plot_api.js

```diff
@@ -1580,6 +1580,12 @@ function _restyle(gd, aobj, _traces) {
                 var moduleAttrs = (contFull._module || {}).attributes || {};
                 var valObject = Lib.nestedProperty(moduleAttrs, ai).get() || {};
 
+                // if restyling entire attribute container, assume worse case
+                if(!valObject.valType) {
```

Comment:
if an attribute container *did* have a `valType` field, would (or could) the schema validation tests fail?

I agree 💯  it's not worth optimizing this.